frontend/src/pages/modelServing/customServingRuntimes/CustomServingRuntimeContext.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import { Outlet } from 'react-router-dom';
+import {
+  Bullseye,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  Spinner,
+  Title,
+} from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import { TemplateKind } from '~/k8sTypes';
+import { DEFAULT_CONTEXT_DATA } from '~/utilities/const';
+import { ContextResourceData } from '~/types';
+import { useContextResourceData } from '~/utilities/useContextResourceData';
+import useTemplates from './useTemplates';
+
+type CustomServingRuntimeContextType = {
+  refreshData: () => void;
+  servingRuntimeTemplates: ContextResourceData<TemplateKind>;
+};
+
+export const CustomServingRuntimeContext = React.createContext<CustomServingRuntimeContextType>({
+  refreshData: () => undefined,
+  servingRuntimeTemplates: DEFAULT_CONTEXT_DATA,
+});
+
+const CustomServingRuntimeContextProvider: React.FC = () => {
+  const servingRuntimeTemplates = useContextResourceData<TemplateKind>(useTemplates('opendatahub')); // TODO: change to operator namespace
```

Comment:
If someone could lend me a hand here, I cannot recall how to fetch the operator's namespace.

And this could be discussed, should we list in all namespaces? How this would play out with permisionless users trying to fetch the templates? should we include a rolebinding to allow them to read too?

I would like to know your opinion about this.

Reply:
`useNamespaces` -- frontend/src/pages/notebookController/useNamespaces.ts

Reply:
Only list data from the dashboard namespace -- no other templates matter for first iteration.

Reply:
The admin view is by admins -- cluster-admin on k8s... so that's fine.

Maybe we have a "getServingRuntimes" custom backend endpoint that uses the service account today? 🤔 Not sure how sane we can make use of a rolebinding to read the values...

Reply:
I understand that -- but we don't control that rolebinding right? That's offered up my the model serving. We don't _really_ have a mechanism here to grant access to the dashboard resources

Reply:
I guess we could do it on pod/backend startup with the service account -- we need to know what namespace the Dashboard is in. Maybe this is doable with the manifests 🤔

Reply:
yes, that's deployed by the dashboard manifests, is not code related, sorry if my previous comment was misleading, I was talking about adding that rolebinding file in the manifest folder.